cirq/google/api/v2/program_test.py

```diff
@@ -72,3 +72,15 @@ def test_line_qubit_from_proto_id_invalid():
 
 def test_named_qubit_from_proto_id():
     assert v2.named_qubit_from_proto_id('a') == cirq.NamedQubit('a')
+
+
+def test_generic_qubit_from_proto_id():
+    assert v2.qubit_from_proto_id('1_2') == cirq.GridQubit(1, 2)
+    assert v2.qubit_from_proto_id('1') == cirq.LineQubit(1)
+    assert v2.qubit_from_proto_id('a') == cirq.NamedQubit('a')
+
+    # int(1_2_3) = 123
+    assert v2.qubit_from_proto_id('1_2_3') == cirq.LineQubit(123)
```

Comment:
This feels strange because it is ambiguous with grid qubit definition and how python treats string ints with underscores.

My suggestion would be to write this code as 
1) split on '_',
2) If two parts, try int cast and grid qubit
3) If one part, attempt to cast to int and a line qubit.
4) make all other cases named.
This case would then be a named qubit.

Reply:
Done.